From 7d6855cb94d3d641809848267484735ae68295b0 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
Date: Mon, 8 Mar 2010 05:20:47 -0300
Subject: [PATCH 33/42] virtio-serial: pci: Allow MSI to be disabled

RH-Author: Amit Shah <amit.shah@redhat.com>
Message-id: <1268025651-8316-2-git-send-email-amit.shah@redhat.com>
Patchwork-id: 7586
O-Subject: [RHEL6 PATCH v2 1/5] virtio-serial: pci: Allow MSI to be disabled
Bugzilla: 567035
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

Michael noted we don't allow disabling of MSI for the virtio-serial-pci
device. Fix that.

Upstream commit: 7b665b668aa92bf0bba696f085dff87539d95529

Signed-off-by: Amit Shah <amit.shah@redhat.com>
CC: "Michael S. Tsirkin" <mst@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
---
 hw/virtio-pci.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/virtio-pci.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/virtio-pci.c b/hw/virtio-pci.c
index 6ebcfe4..4edd8ab 100644
--- a/hw/virtio-pci.c
+++ b/hw/virtio-pci.c
@@ -499,8 +499,8 @@ static int virtio_serial_init_pci(PCIDevice *pci_dev)
     if (!vdev) {
         return -1;
     }
-    vdev->nvectors = proxy->nvectors ? proxy->nvectors
-                                     : proxy->max_virtserial_ports + 1;
+    vdev->nvectors = proxy->nvectors == -1 ? proxy->max_virtserial_ports + 1
+                                           : proxy->nvectors;
     virtio_init_pci(proxy, vdev,
                     PCI_VENDOR_ID_REDHAT_QUMRANET,
                     PCI_DEVICE_ID_VIRTIO_CONSOLE,
@@ -584,7 +584,7 @@ static PCIDeviceInfo virtio_info[] = {
         .init      = virtio_serial_init_pci,
         .exit      = virtio_exit_pci,
         .qdev.props = (Property[]) {
-            DEFINE_PROP_UINT32("vectors", VirtIOPCIProxy, nvectors, 0),
+            DEFINE_PROP_UINT32("vectors", VirtIOPCIProxy, nvectors, -1),
             DEFINE_PROP_HEX32("class", VirtIOPCIProxy, class_code, 0),
             DEFINE_PROP_UINT32("max_ports", VirtIOPCIProxy, max_virtserial_ports,
                                31),
-- 
1.6.3.rc4.29.g8146

